File: src/main/java/me/sd_master92/customvoting/commands/VoteTopCommand.java
package me.sd_master92.customvoting.commands;

import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.VoteFile;
import me.sd_master92.customvoting.constants.Messages;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public class VoteTopCommand implements CommandExecutor
{
    private final Main plugin;

    public VoteTopCommand(Main plugin)
    {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args)
    {
        if (command.getPermission() != null && sender.hasPermission(command.getPermission()))
        {
            List<VoteFile> topVoters = VoteFile.getTopVoters(plugin);
            if (topVoters.size() > 0)
            {
                List<String> messages = new ArrayList<>();
                for (String message : Messages.VOTE_TOP_COMMAND_FORMAT.getMessages(plugin))
                {
                    if (!message.contains("%PLAYERS%"))
                    {
                        messages.add(message);
                    } else
                    {
                        Map<String, String> placeholders = new HashMap<>();
                        for (VoteFile topVoter :
                                topVoters.stream().limit(5).collect(Collectors.toList()))
                        {
                            placeholders.put("%PLAYER%", topVoter.getName());
                            placeholders.put("%VOTES%", "" + topVoter.getVotes());
                            placeholders.put("%s%", topVoter.getVotes() == 1 ? "s" : "");
                            message = Messages.VOTE_TOP_COMMAND_PLAYERS.getMessage(plugin, placeholders);
                            messages.add(message);
                        }
                    }
                }
                for (String message : messages)
                {
                    sender.sendMessage(message);
                }
            } else
            {
                sender.sendMessage(Messages.VOTE_TOP_COMMAND_NOT_FOUND.getMessage(plugin));
            }
        } else
        {
            sender.sendMessage(Messages.NO_PERMISSION.getMessage(plugin));
        }
        return true;
    }
}

File: src/main/kotlin/me/sd_master92/customvoting/commands/VoteTopCommand.kt
package me.sd_master92.customvoting.commands

import me.sd_master92.core.command.SimpleCommand
import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.enumerations.Message
import me.sd_master92.customvoting.constants.interfaces.Voter
import me.sd_master92.customvoting.getVotesPlaceholders
import me.sd_master92.customvoting.sendText
import me.sd_master92.customvoting.sendTexts
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.stream.Collectors

class VoteTopCommand(private val plugin: CV) : SimpleCommand(plugin, "votetop")
{
    override fun onCommand(sender: CommandSender, args: Array<String>)
    {
        val topVoters = Voter.getTopVoters(plugin)
        if (topVoters.isNotEmpty())
        {
            val messages: MutableList<String> = ArrayList()
            for (message in Message.VOTE_TOP_COMMAND_FORMAT.getMessages(plugin))
            {
                if (!message.contains("%PLAYERS%"))
                {
                    messages.add(message)
                } else
                {
                    for (topVoter in topVoters.stream()
                        .limit(5)
                        .collect(Collectors.toList()))
                    {
                        messages.add(
                            Message.VOTE_TOP_COMMAND_PLAYERS.getMessage(
                                plugin,
                                topVoter.getVotesPlaceholders(plugin)
                            )
                        )
                    }
                }
            }
            sender.sendTexts(messages)
        } else
        {
            sender.sendText(plugin, Message.VOTE_TOP_COMMAND_NOT_FOUND)
        }
    }

    override fun onCommand(player: Player, args: Array<String>)
    {
    }

    init
    {
        withNoPermMessage(Message.NO_PERMISSION.getMessage(plugin))
    }
}